armi/reactor/reactors.py

```diff
@@ -466,7 +463,8 @@ def removeAssembly(self, a1, discharge=True):
         self.remove(a1)
 
         if discharge and self._trackAssems:
-            self.sfp.add(a1)
+            if hasattr(self.parent, "sfp"):
```

Comment:
It is unclear to me why this `if` is needed. I think that `self.parent` (i.e. the reactor object) will always have a `sfp` based on the way that the reactor object is initialized. Is that wrong?

Reply:
This comment also applies to a similar `if` statement below.

Reply:
More importantly, if this `if` statement fails, what then happens to the assembly. I guess it will just poof disappear, without the user knowing about it. Because they have set `trackAssems: True`, this would be unexpected behavior for them that they should probably be warned about.

Reply:
Okay, I added a log statement if the SFP doesn't exist, but `trackAssems` is on.  That's a good call.